### PR TITLE
LSPS1: Update message format and get client working

### LINKED ISSUE #87: LSPS1: Disallow compiler warnings once implementation is finished
In https://github.com/lightningdevkit/lightning-liquidity/pull/86 we denied warnings in CI generally, but allowed them explicitly for LSPS1 as the implementation is not finished and currently hidden behind a `cfg` flag anyways.

We should however make sure to disallow warnings again, fix any remaining ones before we drop the `cfg` flag.

### LINKED ISSUE #144: LSPS1: Accommodate `refund_onchain_address`
See https://github.com/BitcoinAndLightningLayerSpecs/lsp/pull/123

### LINKED ISSUE #146: LSPS1: Deprecate `CANCELLED` state
See https://github.com/BitcoinAndLightningLayerSpecs/lsp/pull/126

### PR DESCRIPTION
**Pull request by tnull**
Closes #87
Closes #144
Closes #146

We refactor LSPS1-related client code and remove the `cfg` gate for these parts. Should be good for review, we can always iterate to improve the API etc. over time.

### CODE REVIEW

**Comment by tnull** [comment]
CI failure on the `beta` channel is unrelated: https://github.com/lightningdevkit/rust-lightning/issues/3424

File: src/lsps1/client.rs
Comment (arik-so) [inline src/lsps1/client.rs:1]: I think a little summary in the commit message of how the client logic is simplified would be quite helpful for future readers not needing to just piece it together.

File: src/lsps1/msgs.rs
Comment (arik-so) [inline src/lsps1/msgs.rs:1]: I think perhaps some more descriptions of the additional changes happening here beyond what's in the commit message would be helpful.

Comment (tnull) [reply src/lsps1/msgs.rs:1]: Reworded the commit to make things a bit more explicit.

File: src/lsps1/client.rs
Comment (tnull) [reply src/lsps1/client.rs:1]: Reworded the commit to make things a bit more explicit.

**Review by arik-so — APPROVED** [review]

**Pull request merged**